src/parser/binary_parser.ml

```diff
@@ -1047,7 +1047,19 @@ let sections_iterate (input : Input.t) =
     else Ok ()
   in
 
-  let indexed_of_list l = List.mapi Indexed.return l in
+  let indexed_of_list l = List.rev @@ List.mapi Indexed.return l in
+
+  (* Types *)
+  let types =
+    let typs =
+      List.map
+        (fun (Bt_raw (_ind, (params, results)) : binary block_type) ->
+          Def_func_t (params, results) )
+        type_section
+    in
```

Comment:
OK so, I think this part should move to `read_type` (that you could rename `read_str_type`).

Reply:
Done https://github.com/OCamlPro/owi/pull/232/commits/30c14a8cd8424584cf63298d52c416afa5b70a3d